src/services/content.ts

```diff
@@ -0,0 +1,344 @@
+ * fetchAndDispatchContent retrieves activity content for an annoucement and then dispatches
+ * it to the redux store.
+ * @param dispatch function used to dispatch to store
+ * @param message broadcast annoucnement containing activity content url
+ * @param blockNumber block number of block containing publication
+ * @param blockIndex index of publication within block
+ * @param batchIndex index of announcment within batch
+ * @throws if content cannot be retrieved
+ */
+const fetchAndDispatchContent = async (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement,
+  blockNumber: number,
+  blockIndex: number,
+  batchIndex: number
+) => {
+  const decoder = new TextDecoder();
+
+  try {
+    const url = decoder.decode((message.url as any) as Uint8Array);
+    const res = await fetch(url);
+    const activityContent = await res.json();
+    dispatchActivityContent(
+      dispatch,
+      message,
+      activityContent,
+      blockNumber,
+      blockIndex,
+      batchIndex
+    );
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+/**
+ * dipatchGraphChange updates the social graph in redux with a graph change announcment.
+ * @param dispatch function used to dispatch to store
+ * @param graphChange graph change announcment
+ */
+const dispatchGraphChange = (
+  dispatch: Dispatch,
+  graphChange: GraphChangeAnnouncement
+) => {
+  const decoder = new TextDecoder();
+  dispatch(
+    upsertGraph({
+      follower: decoder.decode((graphChange.fromId as any) as Uint8Array),
+      followee: decoder.decode((graphChange.objectId as any) as Uint8Array),
+      unfollow: graphChange.changeType === DSNPGraphChangeType.Unfollow,
+    })
+  );
+};
+
+/**
+ * handleBatchAnnouncment retrieves and parses a batch and then routes its contents
+ * to the redux store.
+ * @param dispatch function used to dispatch to the store
+ * @param blockIndex index of publication within block
+ */
+const handleBatchAnnouncement = (dispatch: Dispatch) => (
+  batchAnnouncement: BatchPublicationLogData,
+  blockIndex: number
+) => {
+  dsnp.readBatchFile(batchAnnouncement, (announcementRow, batchIndex) => {
+    try {
+      const announcement = announcementRow as unknown;
+      if (dsnp.isGraphChangeAnnouncement(announcement)) {
+        dispatchGraphChange(dispatch, announcement);
+      } else if (dsnp.isBroadcastAnnouncement(announcement)) {
+        fetchAndDispatchContent(
+          dispatch,
+          announcement,
+          batchAnnouncement.blockNumber,
+          blockIndex,
+          batchIndex++
+        );
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  });
+};
+
+/**
+ * storeActivityContent stores activity content on the static server
+ * @param content the content to store
+ * @returns hash of content (from which url can be derived)
+ */
+const storeActivityContent = async (
+  content: ActivityContentNote | ActivityContentProfile
+): Promise<string> => {
+  const hash = keccak256(core.activityContent.serialize(content));
```

Comment:
Likely not for this PR, but the sdk now has hash exposed and it adds the `0x` prefix (although not sure you want that here.)
```suggestion
  const hash = core.utilities.hash(core.activityContent.serialize(content));
```

Reply:
This has to change more drastically because, while hash is now available, serialize is now inaccessible. At any rate, a problem for the SDK PR.